static/app/views/automations/components/dataConditionNodeList.tsx

```diff
@@ -34,9 +44,67 @@ export default function DataConditionNodeList({
   updateCondition,
   updateConditionType,
 }: DataConditionNodeListProps) {
-  const options = Array.from(dataConditionNodesMap.entries())
-    .map(([value, {label}]) => ({value, label}))
-    .filter(({value}) => dataConditionTypes.includes(value));
+  const {data: dataConditionHandlers = []} = useDataConditionsQuery(handlerGroup);
+
+  const options = useMemo(() => {
+    if (handlerGroup === DataConditionHandlerGroupType.WORKFLOW_TRIGGER) {
+      return dataConditionHandlers.map(handler => ({
+        value: handler.type,
+        label: dataConditionNodesMap.get(handler.type)?.label || handler.type,
+      }));
+    }
+
+    const issueAttributeOptions: Option[] = [];
+    const frequencyOptions: Option[] = [];
+    const eventAttributeOptions: Option[] = [];
+
+    dataConditionHandlers.forEach(handler => {
+      const percentageTypes = [
+        DataConditionType.EVENT_FREQUENCY_PERCENT,
+        DataConditionType.PERCENT_SESSIONS_PERCENT,
+        DataConditionType.EVENT_UNIQUE_USER_FREQUENCY_PERCENT,
+      ];
+
+      if (percentageTypes.includes(handler.type)) {
+        return; // Skip percentage types so that frequency conditions are not duplicated
+      }
+
+      const conditionType = frequencyTypeMapping[handler.type] || handler.type;
+
+      const newDataCondition: Option = {
+        value: conditionType,
+        label: dataConditionNodesMap.get(handler.type)?.label || handler.type,
+      };
+
+      if (handler.handlerSubgroup === DataConditionHandlerSubgroupType.EVENT_ATTRIBUTES) {
+        eventAttributeOptions.push(newDataCondition);
+      } else if (handler.handlerSubgroup === DataConditionHandlerSubgroupType.FREQUENCY) {
+        frequencyOptions.push(newDataCondition);
+      } else if (
+        handler.handlerSubgroup === DataConditionHandlerSubgroupType.ISSUE_ATTRIBUTES
+      ) {
+        issueAttributeOptions.push(newDataCondition);
+      }
+    });
+
+    return [
+      {
+        key: DataConditionHandlerSubgroupType.ISSUE_ATTRIBUTES,
+        label: t('Filter by Issue Attributes'),
+        options: issueAttributeOptions,
+      },
+      {
+        key: DataConditionHandlerSubgroupType.FREQUENCY,
+        label: t('Filter by Frequency'),
+        options: Array.from(new Set(frequencyOptions)),
```

Comment:
could maybe just have frequencyOptions as a set

Reply:
oh i forgot to remove this-- this didn't work since i think the options are compared by reference, so none of the duplicates were actually removed